src/attributes.jl

```diff
@@ -188,6 +193,84 @@ function h5readattr(filename, name::AbstractString)
 end
 
 
+struct AttributeDict <: AbstractDict{String,Any}
+    parent::Object
+end
+
+"""
+    attrs(object::Union{File,Group,Dataset,Datatype})
+
+The attributes dictionary of `object`. Returns an `AttributeDict`, a `Dict`-like
+object for accessing the attributes of `object`.
+
+```julia
+attrs(object)["name"] = value  # create/overwrite an attribute
+attr = attrs(object)["name"]   # read an attribute
+delete!(attrs(object), "name") # delete an attribute
+keys(attrs(object))            # list the attribute names
+```
+"""
+function attrs(parent::Object)
+    return AttributeDict(parent)
+end
+attrs(file::File) = attrs(open_group(file, "."))
```

Comment:
```suggestion
AttributeDict(file::File) = AttributeDict(open_group(file, "."))

function attrs(parent)
    return AttributeDict(parent)
end
```

Should the constructor work directly on `File`?

I'm also tempted to suggest `const attrs = AttributeDict`.